endpoints/openrtb2/auction_test.go

```diff
@@ -160,6 +160,91 @@ func TestRefererParsing(t *testing.T) {
 	}
 }
 
+// Test valid/invalid DigiTrust functionality
+func TestImplicitUser(t *testing.T) {
+	for _, requestData := range digiTrustTestRequests {
+		httpReq := httptest.NewRequest("POST", "/openrtb2/auction", strings.NewReader(requestData))
+		bidReq := &openrtb.BidRequest{}
+		err := json.Unmarshal(json.RawMessage(requestData), &bidReq)
+		if err != nil {
+			t.Errorf("Error unmashalling bid request: %s", err.Error())
+		}
+
+		setUserImplicitly(httpReq, bidReq)
+
+		if bidReq.ID == "request-without-user-obj" {
```

Comment:
This looks like a prime spot for a switch/case block.

Reply:
agreed!